### ISSUE TITLE
Run shell commands from MQTT

### ISSUE THREAD

**Issue by nagyrobi** [thanksmister/wallpanel-android#186]
This request is from users running WallPanel on rooted Android devices. 
Having SU and BusyBox installed allows to run shell commands. 
For example [turning off or on](https://github.com/thanksmister/wallpanel-android/issues/179) the screen [is possible](https://android.stackexchange.com/questions/154058/how-to-turn-off-screen-of-android-with-terminal-emulator) on most tablets using a shell command. Also many useful things can be done like starting or closing apps(intents), reboots, etc.

The best would be to simply receive these as MQTT payloads and execute them. 

Add a button in settings "Chek if root is available" (just run "su") that will trigger SuperSU or whatever manager the user has to add WallPanel to the whitelist.

**Comment by thanksmister** [comment]
I think that might be out of scope for this open source project.  Seems like there would need fora  paid version of this application to help support more advanced feature development.  If you look at the feature requests, there are months of work and there are not many contributors.  So a "Pro" version to support more features would seem appropriate at some point.

**Comment by nagyrobi** [comment]
:-(

**Comment by thanksmister** [comment]
We can keep this feature open, it can be useful for a later time.

**Comment by nagyrobi** [comment]
Please [consider](https://github.com/thanksmister/wallpanel-android/issues/187#issuecomment-623516173) this feature request to be part of the open source functionality (as it is targeted to devices running openness).

**Comment by thanksmister** [comment]
This has been added to the backlog of feature requests for WallPanel.

**Comment by nagyrobi** [comment]
Thank you.

**Comment by thanksmister** [comment]
> Add a button in settings "Chek if root is available" (just run "su") that will trigger SuperSU or whatever manager the user has to add WallPanel to the whitelist.

You have to expand on this better, how do you issue the command on a device with root from an Android tablet?

**Comment by ulno** [comment]
I am seeing @thanksmister been totally swamped with feature requests (yes, it's a great project!), but wouldn't it make sense to revert some of these requests to pointing in the documentation to running ssh or your own mosquitto_sub-based script on termux? I have several tablets and old phones where wallpanel and termux co-exist quite happily. And if your device is rooted with busybox anyways, you already have sshd.

Having a well working web-panel with an option of a screen-saver (guess I am preferring some more "unix-philosophy" here) seems to me quite perfect already. Most other features can easily be run from a script in termux (or natively through ssh  in busybox).


Just wondering.

I even wonder if thanksmister producing or gathering central case-based documentation for that and making this (only?) available for potential patreon-supporters would be a good way to some paid contributions to this project?

**Comment by thanksmister** [comment]
@ulno Yeah, termux is a good solution, you don't even root to run termux.  You can run your own broker or even Node-Red with termux.   I think user's want this app to do just about everything, as you can see from the feature list, haha.   

Might be nice to have some documentation for this project for power users on how they can achieve some of these features in other ways.   I am not worried about money as much as time.  I cannot possibly handle all the feature requests. 

I am glad you chimed it, it did not occur to me that there are a lot of features that could be handled by this approach.   I feel like many people request features that duplicate features they can install with other applications as well.

**Comment by thanksmister** [comment]
For those interested in running Node-Red on Android, here is a guide https://nodered.org/docs/getting-started/android.   The application in the Play store actually handles motion detection and a slew of other interactions with the device.

**Comment by thanksmister** [comment]
@ulno I think what I will do is start listing things that can be done through other apps or through Termux or rooted options as either "documentation required" instead of feature requests.   This way, I can capture these types of requests and hopefully get contributes to update the docs with power feature options.  Thanks again for your input.
